images/remote-image.proto

```diff
@@ -7,10 +7,7 @@ message local_image_entry {
 }
 
 message remote_image_entry {
-	required string name		= 1;
-	required string snapshot_id	= 2;
-	required uint32 open_mode	= 3;
-	required uint64 size		= 4;
+	/* No longer used */
 }
```

Comment:
Yes.

I am also tempted to do the following:
1) Remove the snapshot_id logic. This was originally used for incremental checkpoints support. However, the ids are not meaningful. Also, there should be an efficient mechanism to inform the presence of pages that have not changed. This mechanism should also be thought out with filesystem dedup that would be done by the caller. 
2) Rename the "remote" image naming to "streaming" image. The remote name can be confusing with the lazy page server, which is also remote. It would also be consistent with the naming of criu-image-streamer tool.
3) Remove the code paths where `opts.remote` is tested to avoid using `sendfile()` in `criu/util.c` and `criu/files-reg.c`. There's no reason not to use read+write loops when `sendfile()` should be just fine.

what do you think?

Reply:
SGTM. We need to run tests. I think zdtm.py has the --remote option for this. Could you fix it to use criu-image-streamer?